game/main.go

```diff
@@ -0,0 +1,292 @@
+package main
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+)
+
+type Location struct {
+	items       []map[string][]string
+	motions     []string
+	name        string
+	description string
+	access      bool
+}
+
+type Player struct {
+	items    []string
+	missions []string
+}
+
+type World struct {
+	Player          Player
+	CurrentLocation Location
+}
+
+var w = World{}
+var Locations []Location
+
+func initGame() {
+	kitchen := Location{
+		name:        "кухня",
+		description: "ты находишься на кухне",
+		items: []map[string][]string{
+			{
+				"на столе": {"чай"},
+			},
+		},
+		motions: []string{"коридор"},
+		access:  true,
+	}
+
+	hall := Location{
+		name:        "коридор",
+		description: "ничего интересного",
+		items:       nil,
+		motions:     []string{"кухня", "комната", "улица"},
+		access:      true,
+	}
+
+	room := Location{
+		name:        "комната",
+		description: "ты в своей комнате",
+		items: []map[string][]string{
+			{
+				"на столе": {"ключи", "конспекты"},
+			},
+			{
+				"на стуле": {"рюкзак"},
+			},
+		},
+		motions: []string{"коридор"},
+		access:  true,
+	}
+
+	outside := Location{
+		name:        "улица",
+		description: "на улице весна",
+		motions:     []string{"домой"},
+		access:      false,
+	}
+
+	Locations = []Location{kitchen, hall, room, outside}
+
+	w = World{
+		Player: Player{
+			missions: []string{"собрать рюкзак", "идти в универ"},
+		},
+		CurrentLocation: kitchen,
+	}
+
+}
+
+func isMapEmpty(items []map[string][]string) bool {
+	for _, item := range items {
+		if len(item) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func deleteItem(items *[]map[string][]string, name string) {
+
+	for _, itemPlaces := range *items {
+		for place, items := range itemPlaces {
+			var newValue []string
+			found := false
+
+			for _, item := range items {
+				if item != name {
+					newValue = append(newValue, item)
+				} else {
+					found = true
+				}
+			}
+
+			if found {
+				if len(newValue) == 0 {
+					delete(itemPlaces, place)
+				} else {
+					itemPlaces[place] = newValue
+				}
+			}
```

Comment:
Попробуй переписать код и разобрать такое удаление элемента. if else убираем по возможности для меньшей вложенности
```suggestion
			for id, item := range placeItems {
				if item == name {
					placeItems = append(placeItems[:id], placeItems[id+1:]...)
				}
				itemPlaces[place] = placeItems

				if len(placeItems) == 0 {
					delete(itemPlaces, place)
				}
			}
```